dojo.py

```diff
@@ -18,93 +19,85 @@ def create_room(self, room_type, room_names):
 		for name in room_names:
 			room_name = name.lower()
 			room_type = room_type.lower()
-			occupants = []
-			room_capacity = (6 if room_type=='office' else 4)
 
-			current_rooms = []
-			for room in self.rooms:
-				current_rooms.append(room['room_name'])
-			if room_name in current_rooms:
+			found = False
+			for room in self.all_rooms:
+				if room_name == room.room_name:
+					found = True
+			if found:
 				msg += "Room with name " + room_name + " already exists \n"
 			else:
 				if room_type in ('office', 'living'):
-					self.rooms.append({'room_name': room_name, 'occupants': occupants, 'type': room_type, 'room_capacity': room_capacity})
-					if room_type == 'office':
-						room = Office(room_name)
-						self.offices.append({'room_name': room_name, 'occupants': occupants, 'room_capacity': room_capacity})
-						msg += "Office " + room_name + " added successfully \n"
-					elif room_type == 'living':
-						room = LivingSpace(room_name)
-						self.livingSpaces.append({'room_name': room_name, 'occupants': occupants, 'room_capacity': room_capacity})
-						msg += "Living space " + room_name + " added successfully \n"
+					room = Room(room_name)
```

Comment:
Room is an abstract class. This means that it cannot be made on its on. For example, when God creates a human being, he creates a a male or female. We can't have just a person. Same thing with room. A room must either be LivingSpace or Office. It can't just be room.